compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1425,59 +1425,26 @@ fn check_where_clauses<'tcx>(wfcx: &WfCheckingCtxt<'_, 'tcx>, span: Span, def_id
     let tcx = wfcx.tcx();
 
     let predicates = tcx.predicates_of(def_id.to_def_id());
-    let generics = tcx.generics_of(def_id);
-
-    let is_our_default = |def: &ty::GenericParamDef| match def.kind {
-        GenericParamDefKind::Type { has_default, .. }
-        | GenericParamDefKind::Const { has_default, .. } => {
-            has_default && def.index >= generics.parent_count as u32
```

Comment:
`def.index >= generics.parent_count` always holds for own params (see below), no idea why it was added